### PR TITLE
kubeadm: add tests in ./cmd/phases/init/addons_test.go

### PR DESCRIPTION
**Pull request by tukwila**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
[UT]Increase ut for kubeadm phases init addon.go

before this ut:
image

After this ut:

image

That means addons.go ut coverage can increase from 0% to 19.4%.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
thanks for the PR.
while this can increase the code coverage %, i don't think it contributes much to the code stability.

but happy to hear good reasoning on why this can benefit us.

File: cmd/kubeadm/app/cmd/phases/init/addons_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2018 The Kubernetes Authors.
```
Comment (chendave) [inline cmd/kubeadm/app/cmd/phases/init/addons_test.go:2]: 2023 pls

**Review by chendave — COMMENTED** [review]
/assign

Haven't review in detail, but more testcase for better coverage is welcome IMO.

File: cmd/kubeadm/app/cmd/phases/init/addons_test.go (hunk 1 above)
Comment (tukwila) [reply cmd/kubeadm/app/cmd/phases/init/addons_test.go]: yes, correct. -:)

**Comment by tukwila** [comment]
> thanks for the PR. while this can increase the code coverage %, i don't think it contributes much to the code stability.
> 
> but happy to hear good reasoning on why this can benefit us.

yes, i append ut coverage result in description.

File: cmd/kubeadm/app/cmd/phases/init/addons_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+func TestGetAddonPhaseFlags(t *testing.T) {
+	tests := []struct {
+		casename string
+		name     string
+		want     []string
+	}{
+		{
+			casename: "test name is all",
+			name:     "all",
```
Comment (chendave) [inline cmd/kubeadm/app/cmd/phases/init/addons_test.go:34]: remove the filed, and rename the `casename` to `name`.

File: cmd/kubeadm/app/cmd/phases/init/addons_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+				options.DryRun,
+				options.FeatureGatesString,
+				options.NetworkingDNSDomain,
+				options.NetworkingServiceSubnet,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := getAddonPhaseFlags(tt.name)
```
Comment (chendave) [inline cmd/kubeadm/app/cmd/phases/init/addons_test.go:77]: this just covered the func of `getAddonPhaseFlags` which is not an exported func, ideally, we want to have testcase to cover the exported func, e.g. `NewAddonPhase` func instead.

Can you pls create another one for that? here is the example you can follow, 
https://github.com/kubernetes/kubernetes/blob/5c72df728125530f15a519dacc9c308084dc586b/cmd/kubeadm/app/cmd/phases/init/certs_test.go#L42

**Comment by chendave** [comment]
can you pls rename the subject to "kubeadm: xxx" instead.

I am personally +1 to have a testcase for the `addons.go`, but I think we can furtherly increase the coverage here.

**Comment by neolit123** [comment]
> can you pls rename the subject to "kubeadm: xxx" instead.
> 
> I am personally +1 to have a testcase for the `addons.go`, but I think we can furtherly increase the coverage here.

thanks for having a look.
/approve

**Comment by neolit123** [comment]
/retitle kubeadm: add tests in ./cmd/phases/init/addons_test.go

File: cmd/kubeadm/app/cmd/phases/init/addons_test.go (hunk 2 above)
Comment (tukwila) [reply cmd/kubeadm/app/cmd/phases/init/addons_test.go]: yes, corrected.

File: cmd/kubeadm/app/cmd/phases/init/addons_test.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	"reflect"
+	"testing"
+
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+)
+
+func TestGetAddonPhaseFlags(t *testing.T) {
+	tests := []struct {
+		name string
+		want     []string
```
Comment (chendave) [inline cmd/kubeadm/app/cmd/phases/init/addons_test.go:29]: pls run `hack/verify-gofmt.sh` to fix those lint error

Comment (tukwila) [reply cmd/kubeadm/app/cmd/phases/init/addons_test.go]: yes, corrected. Thks.

**Comment by chendave** [comment]
hmm... the unit failure is not false negative, need fix the unit as well.

**Comment by tukwila** [comment]
> hmm... the unit failure is not false negative, need fix the unit as well.

File: cmd/kubeadm/app/cmd/phases/init/addons_test.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+		}, {
+			name: "invalid_name",
+			want: []string{options.CfgPath,
+				options.KubeconfigPath,
+				options.KubernetesVersion,
+				options.ImageRepository,
+				options.DryRun,
+			},
+		},	
+	}
```
Comment (chendave) [inline cmd/kubeadm/app/cmd/phases/init/addons_test.go:78]: lint failure around here, tips, you can always run `gofmt -s -w cmd/kubeadm/app/cmd/phases/init/addons_test.go` to fix this.

Comment (tukwila) [reply cmd/kubeadm/app/cmd/phases/init/addons_test.go:78]: yes, it's my fault -:( 
now, i corrected.

**Review by chendave — APPROVED** [review]
/lgtm

thanks!

**Pull request merged**